src/liballoc/string.rs

```diff
@@ -67,12 +63,12 @@ use crate::vec::Vec;
 /// contents of the string. It has a close relationship with its borrowed
 /// counterpart, the primitive [`str`].
 ///
-/// [`str`]: ../../std/primitive.str.html
-///
 /// # Examples
 ///
 /// You can create a `String` from a literal string with [`String::from`]:
 ///
+/// [`String::from`]: From::from
```

Comment:
Let's link to the relevant impl, not just the trait (this is `&'_ str`)

```suggestion
/// [`String::from`]: #impl-From<%26%27_%20str>
```

Reply:
the exact impl doesn't have special docs and the point is to _reduce_ the direct links :)

Reply:
I think linking to the type is helpful though. It's not clear that 'literal string' means `&'_ str`, especially to beginners.

Also, same comment as below about fragments loading faster than different pages.

Reply:
We can just fix that with an `&str` link :)